lib/pbench/test/unit/server/query_apis/commons_get.py

```diff
@@ -0,0 +1,387 @@
+from http import HTTPStatus
```

Comment:
This seems like a bad idea to clone 387 lines of one file just to add support for GET.

Reply:
The `Commons` base class is already pretty complicated and messy. Some of it is already driven by introspecting a constructed resource object; and extending that would make it "straightforward" if not necessarily trivial to use the resource `SchemaSet` to figure out which HTTP methods are supported by the resource, as well as which parameters go with each.

The question was whether it was worth trying to accomplish that *before* getting the `/dataset/contents` API fixed. At the time, we decided not to complicate this work or create a new dependency.

On one hand, I'd like to see `Commons` cleaned up, generalized, extended, and probably broken down to be used at higher and lower levels in the hierarchy.

On another hand, I'd rather devote that effort to stripping out the flask and request mocks and making the API suite true unit tests.

And all of this seemed extra baggage for this specific PR.

Reply:
There has to be some kind of middle ground here.

If one were to `diff` the new `commons_get.py` and `commons.py`, what changed?  Can't we just pass a new parameter to use `GET` instead?